include/boost/url/detail/replacement_id_rule.hpp

```diff
@@ -0,0 +1,88 @@
+//
+// Copyright (c) 2022 Alan de Freitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/url
+//
+
+#ifndef BOOST_URL_DETAIL_REPLACEMENT_ID_RULE_HPP
+#define BOOST_URL_DETAIL_REPLACEMENT_ID_RULE_HPP
+
+#include <boost/url/detail/config.hpp>
+#include <boost/url/error.hpp>
+#include <boost/url/string_view.hpp>
+#include <boost/url/grammar/alnum_chars.hpp>
+#include <boost/url/grammar/alpha_chars.hpp>
+#include <boost/url/grammar/delim_rule.hpp>
+#include <boost/url/grammar/lut_chars.hpp>
+#include <boost/url/grammar/optional_rule.hpp>
+#include <boost/url/grammar/token_rule.hpp>
+#include <boost/url/grammar/tuple_rule.hpp>
+#include <boost/url/grammar/unsigned_rule.hpp>
+#include <boost/url/grammar/variant_rule.hpp>
```

Comment:
This seems like quite a lot of includes, are you certain every one of them is necessary?